python/pyspark/sql/plot/core.py

```diff
@@ -299,3 +300,28 @@ def area(self, x: str, y: str, **kwargs: Any) -> "Figure":
         >>> df.plot.area(x='date', y=['sales', 'signups', 'visits'])  # doctest: +SKIP
         """
         return self(kind="area", x=x, y=y, **kwargs)
+
+    def pie(self, x: str, y: str, **kwargs: Any) -> "Figure":
+        """
+        Generate a pie plot.
+
+        A pie plot is a proportional representation of the numerical data in a
```

Comment:
nit: should we check the schema before actually computation? e.g. the `y` should be a numerical datatype

Reply:
Good idea! Added.